receiver/otlpreceiver/internal/errors/errors.go

```diff
@@ -0,0 +1,23 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package errors // import "go.opentelemetry.io/collector/receiver/otlpreceiver/internal/errors"
+
+import (
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"go.opentelemetry.io/collector/consumer/consumererror"
+)
+
+func GetStatusFromError(err error) *status.Status {
+	s, ok := status.FromError(err)
+	if !ok {
+		code := codes.Unavailable
```

Comment:
I believe the original PR used `Unavailable` instead of `Unknown` because the default state was to return a retryable error unless the error was permanent, and `Unknown` is not retryable according to OTLP: https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#failures

Reply:
Then it makes sense to at leas comment this for others like me that will ask for same thing. Thanks for clarifying.

Then should we use "Unknown" for Permanent? Maybe a separate PR though.

Reply:
Added a comment but I agree that changes to the code can be its own PR.